packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -209,7 +209,13 @@ class AppBar extends StatelessWidget {
     toolBarRow.add(new Flexible(
       child: new Padding(
         padding: new EdgeInsets.only(left: 8.0),
-        child: title != null ? new DefaultTextStyle(style: centerStyle, child: title) : null
+        child: title != null ?
+          new DefaultTextStyle(
+            style: centerStyle,
+            isSingleLine: true,
```

Comment:
isSingleLine doesn't afffect handling of newlines.  Only soft line wraps are affected.